torch/utils/_pytree.py

```diff
@@ -1476,3 +1497,10 @@ def tree_map_with_path(
 def keystr(kp: KeyPath) -> str:
     """Given a key path, return a pretty-printed representation."""
     return "".join([str(k) for k in kp])
+
+
+def key_get(obj: Any, kp: KeyPath) -> Any:
+    """Given an object and a key path, return the value at the key path."""
+    for k in kp:
+        obj = k.access(obj)
+    return obj
```

Comment:
bikeshedding: the API is called "key_get", but the accessors are called "access". Can we make them  more consistent?

Reply:
ideas:
- access_with_key/access ?
- key_get/get ?

Eh, key_get / access is growing on me some more.

Reply:
lol I hate API design